app/services/vehicle_age_calculator.rb

```diff
@@ -0,0 +1,12 @@
+class VehicleAgeCalculator
+  def initialize(purchase_date, calculation_date)
+    @purchase_date = Date.parse(purchase_date)
+    @calculation_date = calculation_date
```

Comment:
It seems strange to have to parse `purchase_date` but not `calculation_date`.
Ideally, I think it would make sense to parse everything at the beginning of the request and then we only deal with ruby objects/values

Reply:
I think we agreed to leave this for the time being.

Reply:
Ok, have implemented Rob's solution to change them all to dates on the way in.